dbt_semantic_interfaces/model_transformer.py

```diff
@@ -15,24 +17,36 @@
 logger = logging.getLogger(__name__)
 
 
-class PydanticSemanticManifestTransformer:
+class SemanticManifestTransformer(Protocol[SemanticManifestT]):
     """Helps to make transformations to a model for convenience.
 
     Generally used to make it more convenient for the user to develop their model.
     """
 
-    @staticmethod
+    @abstractmethod
     def transform(
-        model: PydanticSemanticManifest,
+        self,
+        model: SemanticManifestT,
         ordered_rule_sequences: Optional[Sequence[Sequence[SemanticManifestTransformRule]]] = None,
-    ) -> PydanticSemanticManifest:
+    ) -> SemanticManifestT:
         """Copies the passed in model, applies the rules to the new model, and then returns that model.
 
         It's important to note that some rules need to happen before or after other rules. Thus rules
         are passed in as an ordered tuple of rule sequences. Primary rules are run first, and then
         secondary rules. We don't currently have tertiary, quaternary, or etc currently, but this
         system easily allows for it.
         """
+        pass
+
+
+class PydanticSemanticManifestTransformer:
+    """Transforms PydanticSemanticManifest."""
+
+    @staticmethod
+    def transform(  # noqa: D
+        model: PydanticSemanticManifest,
+        ordered_rule_sequences: Optional[Sequence[Sequence[SemanticManifestTransformRule]]] = None,
+    ) -> PydanticSemanticManifest:
```

Comment:
This is all good enough for our purposes, so as long as we aren't in some wonky generic typing implicit Any situation I guess we're ok.